src/SprykerEco/Zed/AmazonPay/Communication/Plugin/Oms/Command/AuthorizeOrderCommandPlugin.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * Apache OSL-2
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\AmazonPay\Communication\Plugin\Oms\Command;
+
+use Generated\Shared\Transfer\AmazonpayCallTransfer;
+use Orm\Zed\Sales\Persistence\SpySalesOrder;
+use Spryker\Zed\Oms\Business\Util\ReadOnlyArrayObject;
+use SprykerEco\Shared\AmazonPay\AmazonPayConfig;
+
+/**
+ * @method \SprykerEco\Zed\AmazonPay\Business\AmazonPayFacadeInterface getFacade()
+ */
+class AuthorizeOrderCommandPlugin extends AbstractAmazonpayCommandPlugin
+{
+    /**
+     * {@inheritdoc}
+     * - Uses API adapter for sending a request to Amazon for order authorization operation.
+     *
+     * @api
+     *
+     * @param array $salesOrderItems
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrder $orderEntity
+     * @param \Spryker\Zed\Oms\Business\Util\ReadOnlyArrayObject $data
+     *
+     * @return array
+     */
+    public function run(array $salesOrderItems, SpySalesOrder $orderEntity, ReadOnlyArrayObject $data): array
+    {
+        $amazonpayCallTransfers = $this->groupSalesOrderItemsByAuthId($salesOrderItems);
+        $customerEmail = $orderEntity->getEmail() ?? $orderEntity->getCustomer()->getEmail();
```

Comment:
I don't see a reason to use `??` instead of `?:` here, but this is not a blocker